core/app/models/spree/stock_location.rb

```diff
@@ -99,6 +101,16 @@ def fill_status(variant, quantity)
       end
     end
 
+    def self.default
+      Rails.cache.fetch("#{Rails.application.class.parent_name.underscore}_default_stock_location") do
+        default_location = where(default: true).first
+        if default_location.blank? && (default_location = first)
```

Comment:
Writing lvars from conditionals is IMO an antipattern, as its easily missed by reviews. What about refactoring the read to `first` out  into another lvar (ideally kill this whole branch via a data migration as I flagged in: https://github.com/spree/spree/pull/6935/files#r50870863).